src/reflection.jl

```diff
@@ -39,51 +39,90 @@ function highlight(io::IO, code, lexer)
     return
 end
 
+#
+# Compat shims
+# 
+
+@inline function typed_signature(@nospecialize(job::CompilerJob))
+    u = Base.unwrap_unionall(job.source.tt)
+    return Base.rewrap_unionall(Tuple{job.source.f, u.parameters...}, job.source.tt)
+end
+
+function method_instances(@nospecialize(tt::Type), world::UInt=Base.get_world_counter())
+    return map(Core.Compiler.specialize_method, method_matches(tt; world))
+end
+
+# FIXME: This is a horrible hack, is there a better way?
+function mi_to_func(mi::Core.MethodInstance)
```

Comment:
@vtjnash there is no better way for this right? We want to go from `Tuple{typeof(f), Int}` to `f, (Int,)`

Reply:
In general, the transform is unsound, so it doesn't exist. But there is tuple-type-tail perhaps

Reply:
Right but `Base.tuple_type_tail` and `Base.tuple_type_head` will return `typeof(+)` and not the instance `+` which reflection requires.